darwin/dataset/utils.py

```diff
@@ -0,0 +1,291 @@
+        List of indices of the images for each split
+    """
+    # Expand the list of files with all the classes
+    expanded_list = [(k, c) for k, v in idx_to_classes.items() for c in v]
+    # Stratify
+    file_indices, labels = zip(*expanded_list)
+    X_train, X_tmp, y_train, y_tmp = remove_cross_contamination(
+        *train_test_split(np.array(file_indices), np.array(labels),
+                          test_size=val_percentage + test_percentage,
+                          random_state=split_seed,
+                          stratify=labels)
+    )
+    X_val, X_test, y_val, y_test = remove_cross_contamination(
+        *train_test_split(X_tmp, y_tmp,
+                          test_size=test_percentage / (val_percentage + test_percentage),
+                          random_state=split_seed,
+                          stratify=y_tmp)
+    )
+    # Remove duplicates within the same set
+    # NOTE: doing that earlier (e.g. in remove_cross_contamination()) would produce mathematical
+    # mistakes in the class balancing between validation and test sets.
+    return list(set(X_train)), list(set(X_val)), list(set(X_test))
+
+def split_dataset(
+        dataset,
+        val_percentage: Optional[float] = 0.1,
+        test_percentage: Optional[float] = 0.2,
+        force_resplit: Optional[bool] = False,
+        split_seed: Optional[int] = 42,
+):
+    """
+    Given a local a dataset (pulled from Darwin) creates lists of file names
+    for each split for train, validation, and test.
+
+    Parameters
+    ----------
+    dataset : TODO
+        Dataset to split
+    val_percentage : float
+        Percentage of images used in the validation set
+    test_percentage : float
+        Percentage of images used in the test set
+    force_resplit : bool
+        Discard previous split and create a new one
+    split_seed : in
+        Fix seed for random split creation
+
+    Returns
+    -------
+    splits : dict
+        Keys are the different splits (random, tags, ...) and values are the relative file names
+    """
+    assert dataset is not None
+    annotation_path = Path(dataset.local_path / "annotations")
+    assert annotation_path.exists()
+    annotation_files = list(annotation_path.glob("*.json"))
+
+    # Extract list of classes and create respective files
+    lists_path = dataset.local_path / "lists"
+    lists_path.mkdir(parents=True, exist_ok=True)
+    idx_to_classes_polygon = make_class_list(
+        "classes_polygon.txt", annotation_files, lists_path, "polygon", force_resplit, add_background=True
+    )
+    idx_to_classes_tags = make_class_list(
+        "classes_tags.txt", annotation_files, lists_path, "tag", force_resplit
+    )
+
+    # Create split id, path and final split paths
+    assert val_percentage is not None
+    assert test_percentage is not None
+    assert split_seed is not None
+    assert 0 < val_percentage < 1.0
+    assert 0 <= test_percentage < 1.0
+    assert val_percentage + test_percentage < 1.0
```

Comment:
Since `val_percentage`, `test_percentage` and `split_seed` come from the user we shoudn't assert them but instead raise a ValueException

Reply:
Done